policykit/integrations/discourse/models.py

```diff
@@ -0,0 +1,266 @@
+from django.db import models
+from policyengine.models import Community, CommunityUser, PlatformAction, StarterKit, ConstitutionPolicy, Proposal, PlatformPolicy, CommunityRole
+from django.contrib.auth.models import Permission, ContentType, User
+import urllib
+from urllib import parse
+import urllib.request
+import base64
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+DISCOURSE_ACTIONS = [
+                    'discoursecreatetopic',
+                    'discoursecreatepost'
+                  ]
+
+DISCOURSE_VIEW_PERMS = ['Can view discourse create topic', 'Can view discourse create post']
+
+DISCOURSE_PROPOSE_PERMS = ['Can add discourse create topic', 'Can add discourse create post']
+
+DISCOURSE_EXECUTE_PERMS = ['Can execute discourse create topic', 'Can execute discourse create post']
+
+class DiscourseCommunity(Community):
+    platform = "discourse"
+
+    team_id = models.CharField('team_id', max_length=150, unique=True)
+    api_key = models.CharField('api_key', max_length=100, unique=True)
+
+    def notify_action(self, action, policy, users=None, template=None, channel=None):
+        from discourseintegration.views import post_policy
+        post_policy(policy, action, users, template, channel)
+
+    def save(self, *args, **kwargs):
+        super(DiscourseCommunity, self).save(*args, **kwargs)
+
+        content_types = ContentType.objects.filter(model__in=DISCOURSE_ACTIONS)
+        perms = Permission.objects.filter(content_type__in=content_types, name__contains="can add ")
+        for p in perms:
+            self.base_role.permissions.add(p)
+
+    def make_call(self, url, values=None, action=None, method=None):
+        data = None
+
+        if values:
+            data = urllib.parse.urlencode(values).encode('utf-8')
+
+        call_info = team_id + url
+
+        if method:
+            req = urllib.request.Request(call_info, data, method=method)
+        else:
+            req = urllib.request.Request(call_info, data)
+        req.add_header('Content-Type', 'application/x-www-form-urlencoded')
+        req.add_header('User-Api-Key', api_key)
```

Comment:
Does this work? It isn't working for me when I try to curl using the api key from metagov discourse. The [docs](https://docs.discourse.org/) say to use `Api-Key` and `Api-Username` headers, which work for me locally:
```
curl -X GET "https://discourse.metagov.org/directory_items.json?period=all&order=days_visited" -H "Api-Key: [redacted]" -H "Api-Username: system"
```

Reply:
Tested it out on our local PolicyKit server and it works for me. The auth setup generates API keys automatically for the user rather than requiring them to make one on the Discourse forum. The API keys generated by the setup auth are slightly different from the manually generated ones; they are tagged with user information, so adding in the username is unnecessary for them.